File: euler_2d/cylinder/mapc2p.py
import numpy as np

rin = 1.0         # radius of cylinder
rout = 50.0 * rin # radius of outer boundary

def mapc2p(xc,yc):
    r = rin * np.exp(xc * np.log(rout/rin))
    theta = 2.0 * np.pi * yc
    xp = r * np.cos(theta)
    yp = r * np.sin(theta)
    return xp, yp

File: euler_2d/cylinder/mapc2p.f90
subroutine mapc2p(xc, yc, xp, yp)
   implicit none
   include 'param.h'
   real(8),intent(in)    :: xc, yc
   real(8),intent(inout) :: xp, yp
   ! Local variables
   real(8) :: rin, rout, r, theta

   rin = 1.0d0
   rout = 50.0d0 * rin

   r     = rin * dexp(xc * dlog(rout/rin))
   theta = 2.0d0 * M_PI * yc

   xp = r * dcos(theta)
   yp = r * dsin(theta)

end subroutine mapc2p
